pkg/filter/filter.go

```diff
@@ -297,3 +297,178 @@ func matches(d *objectData, o *Options) bool {
 	}
 	return matchesKinds && matchesNS && matchesNames && matchesAnnot && matchesLabels
 }
+
+// ComponentPredicate is a func that returns true for components that match
+// criteria and false otherwise.
+type ComponentPredicate func(*bundle.Component) bool
+
+// Select takes a list of components and a predicate and returns the
+// components that match the predicate.
+func Select(components []*bundle.Component, predicate ComponentPredicate) []*bundle.Component {
+	var out []*bundle.Component
+	for _, component := range components {
+		if predicate(component) {
+			out = append(out, component)
+		}
+	}
+	return out
+}
+
+// UnstructuredKindIn checks if any of the components object.GetKind()
+// are in kinds.
+func UnstructuredKindIn(kinds ...string) ComponentPredicate {
+	return func(component *bundle.Component) bool {
+		for _, obj := range component.Spec.Objects {
+			for _, kind := range kinds {
+				if obj.GetKind() == kind {
+					return true
+				}
+			}
+		}
+		return false
+	}
+}
+
+// UnstructuredNamespaceIn checks if any of the components object.GetNamespace()
+// are in namespaces.
+func UnstructuredNamespaceIn(namespaces ...string) ComponentPredicate {
```

Comment:
Maybe "ObjectNamespaceIn" in components, the unstructured resources are generally referred to as Objects.

Reply:
Done